pkg/checkmarx/checkmarx.go

```diff
@@ -476,6 +501,23 @@ func (sys *SystemInstance) GetPresets() []Preset {
 // UpdateProjectConfiguration updates the configuration of the project addressed by projectID
 func (sys *SystemInstance) UpdateProjectConfiguration(projectID int, presetID int, engineConfigurationID string) error {
 	engineConfigID, _ := strconv.Atoi(engineConfigurationID)
+
+	var projectScanSettings ScanSettings
+	header := http.Header{}
+	header.Set("Content-Type", "application/json")
+	data, err := sendRequest(sys, http.MethodGet, fmt.Sprintf("/sast/scanSettings/%v", projectID), nil, header)
+	if err != nil {
+		// if an error happens, try to update the config anyway
+		sys.logger.Warnf("Failed to fetch scan settings of project %v: %s", projectID, err)
+	} else {
+		// Check if the current project config needs to be updated
+		json.Unmarshal(data, &projectScanSettings)
+		if projectScanSettings.Preset.PresetID == presetID && projectScanSettings.EngineConfiguration.EngineConfigurationID == engineConfigID {
```

Comment:
I cannot recall what is `EngineConfigurationID`. Any reason why `EngineConfigurationID` is also checked (in addition to `PresetID`)?

Reply:
clarified. `EngineConfigurationID` = project configuration in UI